mongo/integration/unified/client_entity.go

```diff
@@ -73,11 +81,15 @@ func newClientEntity(ctx context.Context, em *EntityMap, entityOptions *entityOp
 		// Configure a command monitor that listens for the specified event types. We don't take the IgnoredCommands
 		// option into account here because it can be overridden at the test level after the entity has already been
 		// created, so we store the events for now but account for it when iterating over them later.
-		monitor := &event.CommandMonitor{
+		commandMonitor := &event.CommandMonitor{
 			Started:   entity.processStartedEvent,
 			Succeeded: entity.processSucceededEvent,
 			Failed:    entity.processFailedEvent,
 		}
+		poolMonitor := &event.PoolMonitor{
+			Event: entity.processPoolEvent,
+		}
+		clientOpts.SetMonitor(commandMonitor).SetPoolMonitor(poolMonitor)
```

Comment:
Why do these need to be set earlier now?

Reply:
They don't need to be. It's just easier to read IMO because all of the monitors are declared and set upfront.